### PR TITLE
chore(iast): urllib propagation

### PR DESCRIPTION
**Pull request by avara1986**
- Allow AST to taint urllib
- Fix flask query_string in Flask < 2.0
- Refactor type hinting on `benchmarks/bm/iast_fixtures/str_methods.py`
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 1]
```diff
@@ -821,27 +815,27 @@ def get_random_string_seed(
...
     return a
 
 
-def conditional_escape(a):  # type: (str) -> str
+def conditional_escape(a: str) -> Text:
     return a
 
 
-def format_html(a, args):  # type: (str, *Tuple) -> str
+def format_html(a, args):
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:826]: missing type

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 2]
```diff
@@ -11,20 +11,14 @@
...
-
-if TYPE_CHECKING:  # pragma: no cover
-    from typing import Callable  # noqa:F401
-    from typing import Dict  # noqa:F401
-    from typing import Generator  # noqa:F401
-    from typing import Iterable  # noqa:F401
-    from typing import List  # noqa:F401
-    from typing import Sequence  # noqa:F401
-    from typing import Tuple  # noqa:F401
+from typing import Callable
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:14]: why remove the if TYPE_CHECKING?

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 3]
```diff
@@ -86,16 +80,16 @@ def do_tuple_string_assignment(a):
...
+    def wrapper(a: str, b: str) -> Text:
         func = function(a, b)
         return func.upper()
 
     return wrapper
 
 
 @uppercase_decorator
-def do_add_and_uppercase(a, b):  # type: (str, str) -> str
+def do_add_and_uppercase(a, b) -> Text:
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:92]: Types on a, b missing

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 4]
```diff
@@ -191,7 +185,7 @@ def do_translate(s, translate_dict):
     return s.translate(translate_dict)
 
 
-def do_decode_simple(s: bytes) -> str:
+def do_decode_simple(s: bytes) -> Text:
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:188]: I think for a decode test it's a lot better to check that the return value is explicitly str

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 5]
```diff
@@ -821,27 +815,27 @@ def get_random_string_seed(
...
     return a
 
 
-def format_html(a, args):  # type: (str, *Tuple) -> str
+def format_html(a, args):
     return a.join(args)
 
 
-def format_html_join(attrs, args_generator=None):  # type: (str, List[str]) -> str
+def format_html_join(attrs, args_generator=None):
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:830]: type missing entirely here

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 6]
```diff
@@ -851,99 +845,97 @@ def repeat_text_with_join(text, times=2):  # type: (str, int) -> str
 
 
 def do_format_with_positional_parameter(
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:847]: Types missing on these two

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 7]
```diff
@@ -1015,73 +1003,73 @@ def do_slice_condition(s: str, first, second):
...
 
-def do_namedtuple(s: Text) -> Any:
+def do_namedtuple(s: Text):
     PathInfo = namedtuple("PathInfo", "name surname")
     my_string = PathInfo(name=s, surname=None)
     return my_string
 
 
-def do_split_no_args(s):  # type: (str) -> List[str]
+def do_split_no_args(s):
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:1012]: type missing

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 8]
```diff
@@ -1015,73 +1003,73 @@ def do_slice_condition(s: str, first, second):
...
     return s.split()
 
 
-def do_rsplit_no_args(s):  # type: (str) -> List[str]
+def do_rsplit_no_args(s):
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:1016]: type missing

File: benchmarks/bm/iast_fixtures/str_methods.py [hunk 9]
```diff
@@ -1015,73 +1003,73 @@ def do_slice_condition(s: str, first, second):
...
     return s.rsplit()
 
 
-def do_split_maxsplit(s, maxsplit=-1):  # type: (str, int) -> List[str]
+def do_split_maxsplit(s, maxsplit=-1):
```
Comment (juanjux) [inline benchmarks/bm/iast_fixtures/str_methods.py:1020]: type missing or partially missing on all functions from here to the end of the file

File: ddtrace/appsec/_handlers.py [hunk 10]
```diff
@@ -194,22 +194,22 @@ def _on_request_init(wrapped, instance, args, kwargs):
...
             from ddtrace.appsec._iast.processor import AppSecIastSpanProcessor
 
             if not AppSecIastSpanProcessor.is_span_analyzed():
                 return
 
-            # TODO: instance.query_string = ??
             instance.query_string = taint_pyobject(
                 pyobject=instance.query_string,
-                source_name=OriginType.QUERY,
+                source_name=origin_to_str(OriginType.QUERY),
```
Comment (juanjux) [inline ddtrace/appsec/_handlers.py:206]: this `origin_to_str` should probably be done on the `taint_pyobject` function when getting an origin as `source_name` so we don't have to do it on each call to `taint_pyobject`.

**Review by juanjux — APPROVED** [review]
Some nitpicks about missing types and some convenience tip but LGTM overall.

File: ddtrace/appsec/_handlers.py (hunk 10 above)
Comment (avara1986) [reply ddtrace/appsec/_handlers.py:206]: source name is the key value (that is, a string) in most of the cases, but if we don't have any key we set the origin type

File: benchmarks/bm/iast_fixtures/str_methods.py (hunk 2 above)
Comment (avara1986) [reply benchmarks/bm/iast_fixtures/str_methods.py:14]: cause we don't need it anymore, it's related to type hinting comments

**Pull request merged**